packaging/rpm/microshift.spec

```diff
@@ -62,6 +62,7 @@ Requires: iptables
 Requires: microshift-selinux
 Requires: microshift-networking
 Requires: conntrack-tools
+Requires: sos
```

Comment:
Do we have numbers on the size of sos + dependencies? (On a bare try here on my RHEL it showed 3.2MB which is probably fine everywhere and we get the guarantee if having sos always)

If we wanted to make this "optional" install in some cases we can use a weak dependency:
```
Recommends: sos
```

This way sos could be avoided during installation, but I suspect the specific flag to avoid weak dependencies is not even available in the image builder.




I lean more towards what you just proposed, but wanted to share the thought.

Reply:
IMO, the sos reporting tool is something that belongs into the base R4E installation, not into MicroShift. Unlike in OCP, we do not "own" the OS. *If* we want to make it a dependency, we should definitely make it a weak dependency, so users can opt out of it.

Reply:
The recommended dependency works nicely - thank you for the suggestion.
I'm making sure that RHEL for Edge images pull the recommended dependency by default before uploading the fixed code for review.

Reply:
I checked with the Image Builder team and there's no option to automatically pull in weak RPM dependencies. So, unless we create a strong dependency, we cannot guarantee sos utility presence.

@fzdarsky do you thik it's a good enough reason to have a strong dependency on sos?

Reply:
No, a weak dependency would be pulled in always, but you can opt-out.

With the "Recommends" users would be getting it in image-builder. And maybe at some point, we could get them to allow opt-outs, that's probably something that would need implementation on ostree.

In dnf you would do:
`dnf --setopt=install_weak_deps=False` which I only noticed in the container world where you want a very thin container most of the time.

Reply:
I tested the "Recommends" dependency and Image Builder does not pull the "sos" package when weak dependency is specified. I double checked with the Image Builder team and they confirmed this is the current implementation - the only way is to either add strong dependency, or specify the package in the blueprint.

Reply:
Oh, maybe rpmostree interprets --setopt=install_weak_deps=False by default.

Thanks for testing it in image builder @ggiguash